### DISCUSSION TITLE
Add Integration to Bookshare

### DISCUSSION THREAD

**Discussion by brichwin** [edrlab/thorium-reader#1859] [General]
It would be great to have a meaningful interface for searching Bookshare. 

Allow searching by the usual Author, Title, ISBN, free, accessibility features, etc. The search results could be presented an a dialog with an accessible table that lists significant title info (author, title, edition, etc.) and features (images, quality, etc.). 

Selecting a tile from the search results could provide more title details (like the synopsis, etc.) and provide download options (images yes/no, etc.). 

A download progress/success/failure dialog would be helpful too.

**Comment by gautierchomel** [comment]
I love the idea, to do so we need bookshare to share this info, the best way would be [OPDS](https://opds.io/) as it is already supported by [Thorium catalogs feature](https://edrlab.github.io/thorium-reader-doc/220_organizing/222_catalogs/). CC @clapierre

**Comment by jbrugge** [comment]
Bookshare does not support OPDS at the moment, although it has been talked about at different points in the past. There is a [RESTful API](https://apidocs.bookshare.org/) that's available to anyone, with support for searching, downloading and getting detailed information about any title. Searching and title detail lookup are available to anyone, but downloading most titles would require presenting credentials as an approved Bookshare user. For that we support [OAuth 2](https://apidocs.bookshare.org/auth/index.html).

Hope that helps.
